bundles/http/Citrix/00000019-0000-0000-0000-000000000000/47a6eb95-6f9d-44df-8be7-2322ada5770c.js

```diff
@@ -0,0 +1,849 @@
+                if (i == 0) { //calculating number of pages for pagination
+				    noOfPages = calculateNumberOfPages(+responseData.headers.get('X-Total-Count'))
+			    }
+                if (documentsLists.newChangeId < +json[json.length - 1].changeSequenceId) { //getting the  lastChangeSequenceId for incremental sync from page
+				    documentsLists.newChangeId = +json[json.length - 1].changeSequenceId
+                }
+                documentsLists = parseChangesGetDocIds(json, documentsLists) //creating list on requisition and invoices
+		    }
+	    } else {
+            console.log( 'Warning: unable to get changes: ' + responseData.statusText)
+        }
+    }
+    return documentsLists
+}   
+
+function syncAriba(dataStore, client, context, incrementalSync) 
+{
+	console.log('Ariba sync started')
+        
+    let documentsLists  = { //creating lists od rq and invc ids
+        requisitions: [],
+        invoices: [],
+        newChangeId: lastChangeId
+       }
+    
+    if (incrementalSync) {     
+        if (context.lastChangeSequenceId > lastChangeId) {     // reading stored lastChangeSequenceId for incremental sync
+            lastChangeId = context.lastChangeSequenceId
+        }
+    }
+    else {
+        documentsLists = fetchApprovablesAndGetDocuments(client, documentsLists) //geting documents id, this endpoint is used only in full sync hold documents for open approvals infinitely
+    }
+            
+    documentsLists = fetchChangesGetDocumentsAndGetLastId(client, documentsLists) //geting documents id, this endpoint is used for full and incr, hold all data only for last 90 days 
+
+    storeRequisition(dataStore, client, documentsLists.requisitions) //storing requisitions and resolve group approvables
+          
+    if (documentsLists.newChangeId > lastChangeId ) {context.lastChangeSequenceId = documentsLists.newChangeId}     //saving  lastChangeSequenceId for incremental sync into context
+    
+    console.log('Max Change ID ' + context.lastChangeSequenceId)
+	console.log('Total number of requisitions ' + documentsLists.requisitions.length)
+	console.log('Total number of invoices ' + documentsLists.invoices.length)
+    console.log('Sync finished')
+    return
+}
+
+function setIntegrationParametersGlobal(integrationParameters){
+    apiKey = integrationParameters.apiKey  // please see ariba documentation
+    realm = integrationParameters.realm // please see ariba documentation
+    lastChangeId = integrationParameters.lastChangeId // Ariba internal change id, usually 1, please see ariba documentation
+
+    return
+}
+
+function fullSync ({dataStore, client, context, integrationParameters}) {
+    setIntegrationParametersGlobal(integrationParameters)
+    return syncAriba(dataStore, client, context, false)
+}
+
+function incrSync ({dataStore, client, context, integrationParameters}) {
+    
+    setIntegrationParametersGlobal(integrationParameters)
+    return syncAriba(dataStore, client, context, true)
+}
```

Comment:
```suggestion
function incrSync (params) {
    return syncAriba(params, true)
}
```

Reply:
Changed, passing parameters to all functions.